kv_storage.h

```diff
@@ -0,0 +1,230 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef SHERLOCK_KV_STORAGE_H
+#define SHERLOCK_KV_STORAGE_H
+
+#include <atomic>
+#include <future>
+#include <string>
+
+#include "sherlock.h"
+
+#include "../Bricks/mq/inmemory/mq.h"
+
+namespace sherlock {
+namespace kv_storage {
+
+// TODO(dkorolev): Exception hierarchy for KeyValueStorage.
+struct KeyNotPresentException : bricks::Exception {};
+
+// API calls, `[Async]{Get/Set}(...)`.
+//
+// C++ API supports three syntaxes:
+//
+//   1) Asynchronous, using std::{promise/future}.
+//      To use the async/await paradigm.
+//      `AsyncSet(key, value)` and `AsyncGet(key)`, both return a future. The `Get()` one may throw.
```

Comment:
Is ` The `Get()` one may throw.` left intentionally in the async pattern description?